core/src/main/java/org/infinispan/distexec/spi/DefaultDistributedTaskLifecycle.java

```diff
@@ -9,7 +9,7 @@
 public class DefaultDistributedTaskLifecycle implements DistributedTaskLifecycle {
 
    @Override
-   public <T, K, V> void onPreExecute(Callable<T> task, Cache<K, V> inputCache) {
+   public <T, K, V> void onPreExecute(Callable<T> task, Cache<K, V> inputCache, K... inputKeys) {
```

Comment:
No. It‘s the same with List. Because the inputKeys in DistributedTaskPart is List<Object> and the keys in DistributedExecuteCommand is Set<Object>. It uses different collection container. No differences from array.
Instead, I used Collections.unmodifiableCollection() now.